src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -86,5 +132,53 @@ private Task OnFileWatcher(IFileItem file, object obj)
                 _poll = null;
             }
         }
+
+        private async Task<string> GetLastPollStatusAsync()
+        {
+            try
+            {
+                var apiHubBlob = GetBlobReference();
+
+                return await apiHubBlob.DownloadTextAsync();
+            }
+            catch (StorageException exception)
+            {
+                if (exception.RequestInformation == null ||
+                    exception.RequestInformation.HttpStatusCode != 404)
+                {
+                    var errorText = string.Format("Encountered a storage exception while getting the next poll status from the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, exception.Message);
+                    _trace.Error(errorText);
+                }
+                return null;
+            }
+            catch (Exception e)
+            {
+                var errorText = string.Format("Encountered an exception while getting the next poll status from the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, e.Message);
+                _trace.Error(errorText);
+                return null;
+            }
+        }
+
+        private async Task SetNextPollStatusAsync(string nextPoll)
+        {
+            try
+            {
+                var apiHubBlob = GetBlobReference();
+                await apiHubBlob.UploadTextAsync(nextPoll);
+            }
+            catch (Exception e)
+            {
+                var errorText = string.Format("Encountered an exception while setting the next poll status to the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, e.Message);
+                _trace.Error(errorText);
+            }
+        }
+
+        private CloudBlockBlob GetBlobReference()
+        {
+            // Path to apiHub blob is:
+            // apihubs/{siteName}/{functionName}/{triggerPath}/{connectionStringSetting}
```

Comment:
It seems odd to me that triggerPath/connectionString would be part of the path. Why isn't the function name sufficient? A function maps to a single path/connection. Same as timer trigger - I don't include the timer schedule as part of the path.

I'd expect the path to be `apihubs/{siteName}/{functionName}/status`, using the name "status" for a blob file that contains the status info. Also recommend that you read/write JSON to the file so this is extensible in the future if you need to add more properties. For now it would just have your single property.